tests/checks/symlinks-not-overwritten.fish

```diff
@@ -0,0 +1,45 @@
+#RUN: %fish -C 'set -g fish %fish' %s
+begin
+    # fish_variables
+    set -gx XDG_CONFIG_HOME (mktemp -d)
+    mkdir -p $XDG_CONFIG_HOME/fish
+    set -l target_file $XDG_CONFIG_HOME/fish/target_fish_variables
+    set -l fish_variables $XDG_CONFIG_HOME/fish/fish_variables
+
+    echo >$target_file
+    ln -sf $target_file $fish_variables
+    $fish -c 'set -U variable value'
+
+    if not test -L $fish_variables
+        echo fish_variables has been overwritten
+    else if test (wc -c < $target_file) -le 3
+        echo fish_variables was never written
+    else
+        echo fish_variables is still a symlink
+    end
+    # CHECK: fish_variables is still a symlink
+
+
+    # fish_history
+    set -gx XDG_DATA_HOME $XDG_CONFIG_HOME
+    set -l history_file $XDG_DATA_HOME/fish/fish_history
+    set -l target_file $XDG_DATA_HOME/fish/target_fish_history
+    set -l backup_file $XDG_DATA_HOME/fish/fish_history_backup
+
+    echo '- cmd: echo I will be deleted from history
+            when: 1614577746' >$target_file
+    cp $target_file $backup_file
+    ln -sf $target_file $history_file
+    # The one way to ensure non-interactive fish writes to the history file
+    $fish -c 'echo all | history delete deleted | grep echo'
+    # CHECK: [1] echo I will be deleted from history
+
+    if not test -L $history_file
+        echo fish_history has been overwritten
+    else if test "(cat $target_file)" = "(cat $backup_file)"
```

Comment:
This is always false since the `cat` is not run.

```suggestion
    else if cmp $target_file $backup_file
```

Reply:
That's very good to know. Apart from working, it's also less ugly.

I'll just fix it in my version to rebase one less time.

Reply:
I'll also add `>/dev/null`, unless you have a better idea.

Reply:
Actually, `cmp` gives a message to `stderr`: `cmp: EOF on /tmp/tmp.z101oB6q12/fish/target_fish_history which is empty `. So, it'll be &>/dev/null.